stdlib/public/core/FloatingPoint.swift

```diff
@@ -1768,15 +1768,8 @@ extension BinaryFloatingPoint {
           ~(Self.nan.significandBitPattern |
             Self.signalingNaN.significandBitPattern)
       let payload = Self.RawSignificand(truncatingIfNeeded: payload_) & mask
-      // Although .signalingNaN.exponentBitPattern == .nan.exponentBitPattern,
-      // we do not *need* to rely on this relation, and therefore we do not.
-      let value = source.isSignalingNaN
-        ? Self(
-          sign: source.sign,
-          exponentBitPattern: Self.signalingNaN.exponentBitPattern,
-          significandBitPattern: payload |
-            Self.signalingNaN.significandBitPattern)
-        : Self(
+      let value =
+        Self(
```

Comment:
This is a simplification to the NaN conversion routines borne out of a re-reading of IEEE 754, which in its own language indicates that, although the payload should be preserved when converting between types, a signaling NaN should be silenced to a quiet NaN. This now also matches what the concrete initializers do (on my machine, at least).

Reply:
This is not quite correct. A signaling NaN should be silenced to a quiet NaN _and raise the invalid exception_. If the invalid exception is not raised, then the signalingness should be preserved. Swift doesn't have formal support to query the floating-point exception state yet, so this is all a bit moot, but it's worth being precise about it for the sake of future-proofing.

Reply:
We've been lax about silencing signaling NaNs elsewhere in generic code, hence the simplification here; but since it's fine to preserve the signalingness, then I'll leave it be.

Reply:
Yup, but we can still try to do it right for new code =)